Importer/CsvImporter.php

```diff
@@ -23,88 +27,241 @@ class CsvImporter extends AbstractImporter
     protected $delimiter;
 
     /**
-     * @param ConfigurationInterface   $configuration
-     * @param InterpreterInterface     $interpreter
-     * @param EntityManager            $entityManager
-     * @param CsvParser                $parser
-     * @param string                   $delimiter
+     * @var ConfigurationInterface
+     */
+    protected $configuration;
+
+    /**
+     * @var EntityManager
+     */
+    protected $entityManager;
+
+    /**
+     * @var ImporterErrorHandlerInterface[]
+     */
+    protected $importerErrorHandlers = [];
+
+    /**
+     * @var InterpreterInterface
+     */
+    protected $interpreter;
+
+    /**
+     * @var Log
+     */
+    protected $log;
+
+    /**
+     * @var bool
+     */
+    protected $headers;
+
+    /**
+     * @param ConfigurationInterface $configuration
+     * @param InterpreterInterface   $interpreter
+     * @param EntityManager          $entityManager
+     * @param CsvParser              $parser
+     * @param string                 $delimiter
+     * @param Log                    $log
+     * @param boolean                $headers
      */
     public function __construct(
         ConfigurationInterface $configuration,
         InterpreterInterface $interpreter,
         EntityManager $entityManager,
         CsvParser $parser,
-        $delimiter = ','
-    )
-    {
+        $delimiter = ',',
+        Log $log,
+        $headers = true
+    ) {
         $this->parser = $parser;
-        parent::__construct($configuration, $interpreter, $entityManager);
         $this->delimiter = $delimiter;
+        $this->configuration = $configuration;
+        $this->entityManager = $entityManager;
+        $this->interpreter = $interpreter;
+        $this->log = $log;
+        $this->headers = $headers;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasHeaders()
+    {
+        return $this->headers;
+    }
+
+    /**
+     * @param bool $headers
+     */
+    public function setHeaders($headers)
```

Comment:
Thinking about it, this would read `$csvImporter->hasHeaders()` which might not be so obvious. Perhaps it is better to say something like `$csvImporter->csvHasHeaders()`?